pkg/tfgen/generate_nodejs.go

```diff
@@ -479,6 +479,20 @@ func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (stri
 		w.Writefmtln("")
 	}
 
+	w.Writefmtln("    public static readonly __pulumiType = '%s';", res.info.Tok)
```

Comment:
i would just call this "type".  Though there's a small concern of you colliding with a resource field of that name.  i don't like it being both public and `__`.  either remove the `__` or make private.

Reply:
We decided offline to make this `@internal` and `public`.